src/v2/guide/typescript.md

```diff
@@ -187,3 +187,43 @@ const Component = Vue.extend({
 ```
 
 If you find type inference or member completion isn't working, annotating certain methods may help address these problems. Using the `--noImplicitAny` option will help find many of these unannotated methods.
+
+
+
+## Annotating Props
+
+```ts
+import Vue, { PropType } from 'vue'
+
+interface ComplexMessage { 
+  title: string,
+  okMessage: string,
+  cancelMessage: string
+}
+const Component = Vue.extend({
+  props: {
+    name: String,
+    success: { type: String },
+    callback: { 
+      type: Function as PropType<()=>void>
+    },
+    message: {
+      type: Object as PropType<ComplexMessage>,
+      required: true,
+      validation(message: ComplexMessage){
```

Comment:
```suggestion
      validator (message: ComplexMessage) {
```

Reply:
You can probably use arrow functions here too. I think the main docs don't use them because of browser support but TS has to be transpiled anyway.

Reply:
See https://github.com/vuejs/vue/issues/8679#issuecomment-414437327